### ISSUE TITLE
MAC addresses that begin with F shifts all characters left one position

### ISSUE THREAD

**Issue by abishur** [lupinelab/proxmox_wol#1]
Edit: Original Issue was named "VM does not start", I have changed the title to reflect what I discovered as I tested out different things trying to fix my problem.

I've setup this docker container on my Proxmox node (running 7.2-11) inside an LXC container.  
I have also setup a portainer container and can confirm that docker in general is at least working

I have copied the docker-compose.yaml script you've provide and changed pm_nodes to the name and ip of my proxmox server (I only have the one so I deleted the second line)

I entered the pm user and token name and token that was generated and made sure everyone had the correct permissions

I can watch the logs of the container from portainer and see that it does see the magic packet come in and the mac address is of the VM I want to turn on, but the VM isn't turning on.

I'm not using VLANs, both my VM I want to turn on and my LXC container are using the vmbr0 bridge, the vm is using a VirtIO network card.  When I send the magic packet I see this in my logs:

`Listening for magic packets on port 9:
Heard E9:DE:39:34:F5:1F
Listening for magic packets on port 9:
Heard E9:DE:39:34:F5:1F
Listening for magic packets on port 9:
Heard E9:DE:39:34:F5:1F
Listening for magic packets on port 9:
Heard E9:DE:39:34:F5:1F`

Any ideas what is going on with it?

**Comment by abishur** [comment]
I just noticed that that is not my mac address of the VM.  My mac address should be:

FE:9D:E3:93:4F:51

not

E9:DE:39:34:F5:1F

I've tried to re-arrange my mac address to fix this, but it keeps reporting that it heard the same mac address.  Where is the code it is using to parse the mac address?

**Comment by abishur** [comment]
Okay I've done some more research, any for any mac address that starts with an "F" it will move the leading "F" character to the end of the MAC address.  Every other leading character had no issues, F characters I placed randomly in the address appeared to have no issue, it's only the beginning character that creates an issue if that character is F.

I have edited my issue title to correctly reflect the real issue that was occurring.  Once I changed the mac address to start with a different character I was able to start it without issues.
